src/unit/test_hashset.c

```diff
@@ -420,62 +420,63 @@ int test_safe_iterator(int argc, char **argv, int flags) {
     UNUSED(argv);
     UNUSED(flags);
 
-    long count = 1000;
+    size_t count = 1000;
+    uint8_t element_counts[count * 2];
+    memset(element_counts, 0, sizeof element_counts);
 
     /* A set of longs, i.e. pointer-sized values. */
     hashsetType type = {0};
     hashset *s = hashsetCreate(&type);
-    long j;
 
     /* Populate */
-    for (j = 0; j < count; j++) {
-        assert(hashsetAdd(s, (void *)j));
+    for (size_t j = 0; j < count; j++) {
+        assert(hashsetAdd(s, element_counts + j));
     }
 
     /* Iterate */
-    uint8_t element_returned[count * 2];
-    memset(element_returned, 0, sizeof element_returned);
-    long num_returned = 0;
+    size_t num_returned = 0;
     hashsetIterator iter;
     hashsetInitSafeIterator(&iter, s);
-    while (hashsetNext(&iter, (void **)&j)) {
+    uint8_t *item;
+    while (hashsetNext(&iter, (void **)&item)) {
+        size_t index = item - element_counts;
         num_returned++;
-        if (j < 0 || j >= count * 2) {
-            printf("Element %ld returned, max == %ld. Num returned: %ld\n", j, count * 2 - 1, num_returned);
+        if (item < element_counts || item >= element_counts + count * 2) {
+            printf("Element %zu returned, max == %zu. Num returned: %zu\n", index, count * 2 - 1, num_returned);
             printf("Safe %d, table %d, index %lu, pos in bucket %d, rehashing? %d\n", iter.safe, iter.table, iter.index,
                    iter.pos_in_bucket, !hashsetIsRehashing(s));
             hashsetHistogram(s);
             exit(1);
         }
```

Comment:
This was debugging code that I used while implementing. I don't think we need to keep it. Do you?

I think we can delete it. There's already an assert below:

    assert(item >= element_counts || item < element_counts + count * 2);